src/battle_util.c

```diff
@@ -2062,419 +1962,258 @@ u8 DoFieldEndTurnEffects(void)
                 BattleScriptExecute(gBattlescriptCurrInstr);
                 effect++;
             }
-            gBattleStruct->turnCountersTracker++;
+            gBattleStruct->endTurnEventsCounter++;
+            break;
+        case ENDTURN_WEATHER_FORM: // TODO: Figure out where to handle those
+            if (AbilityBattleEffects(ABILITYEFFECT_ON_WEATHER, battler, 0, 0, 0))
+                effect++;
+            gBattleStruct->turnEffectsBattlerId++;
             break;
-        case ENDTURN_DAMAGE_NON_TYPES:
-            while (gBattleStruct->turnSideTracker < 2)
+        case ENDTURN_WEATHER_DAMAGE: // TODO: Refactor this block to be more readable
+            if (!IsBattlerAlive(battler) || !WEATHER_HAS_EFFECT || ability == ABILITY_MAGIC_GUARD)
             {
-                side = gBattleStruct->turnSideTracker;
-                if (gSideStatuses[side] & SIDE_STATUS_DAMAGE_NON_TYPES && --gSideTimers[side].damageNonTypesTimer == 0)
-                {
-                    // There is no in-game message when this side status ends.
-                    gSideStatuses[side] &= ~SIDE_STATUS_DAMAGE_NON_TYPES;
-                    effect++;
-                }
-                gBattleStruct->turnSideTracker++;
-                if (effect != 0)
-                    break;
+                gBattleStruct->turnEffectsBattlerId++;
+                break;
             }
-            if (!effect)
+            else if ((gBattleWeather & B_WEATHER_SUN && ability == ABILITY_DRY_SKIN)
```

Comment:
**suggestion**: Does it make more sense to organize these under weather-umbrellas? Is that possible without altering overall outcome?

Reply:
yep, I can do that I think